### ISSUE TITLE
[0.50.02] Unable to re-deploy unit to stratcon map if GM-undeployed from TO&E

### ISSUE THREAD

**Issue by PhoenixHeart512** [MegaMek/mekhq#5576]
### Prerequisites and Pre-Issue Checklist

- [X] I'm reporting the issue to the correct repository:
- [MegaMek](https://github.com/MegaMek/megamek/issues)
- [MegaMekLab](https://github.com/MegaMek/megameklab/issues)
- [MekHQ](https://github.com/MegaMek/mekhq/issues)

- [X] I've tested the issue against at least the latest MILESTONE version
- [X] I've asked on the MegaMek Discord about the error
- [X] I've reviewed the BattleTech rules and MekHQ documentation, and I've confirmed that something isn't working as intended.
- [X] I've searched the Github tracker and haven't found the issue listed

### Severity *

Medium (Gameplay Limitation): Non-core functionality is impaired, providing a suboptimal but playable experience.

### Brief Description *

If:
- Force is assigned to a scenario on the Stratcon map
- Player uses GM tools to un-deploy force from TO&E

Then player is unable to re-deploy force back to the scenario because the primary force still shows as deployed to the stratcon map, even though the Briefing Room shows no units assigned to the scenario.

Suggested solution: When a force is undeployed using the GM tools on TO&E tab, have it also removed from the Area of Operations tab map so that a new force (or corrected original force) can be re-deployed to the scenario properly.

### Steps to Reproduce

- Force is assigned to a scenario on the Stratcon map
- Player uses GM tools to un-deploy force from TO&E
- Force now shows as not deployed in TO&E
- Force DOES still show as deployed on Area of Operations tab/map
- Unable to assign new forces to the scenario because the Area of Operations map still thinks a unit has already been deployed

### Operating System *

Windows 10

### Java Version *

Java Version: 17.0.6

### MekHQ Suite Version *

Free Text (type manually)

### Custom MekHQ Version

megamek.MegaMek.initializeLogging(MegaMek.java:150) - Starting MegaMek v0.50.02     Build Date: 2024-12-30T14:34:16.149525500     Today: 2024-12-31     Origin Project: MegaMekLab     Java Vendor: Eclipse Adoptium     Java Version: 17.0.6     Platform: Windows 10 10.0 (amd64)     System Locale: en_US     Total memory available to MegaMek: 8 GB     MM Code Revision: 2bc80c102a745fedbb40039b75202ecdfcf4d8d2     MML Code Revision: 218b264a92737d8dfc7b65952e4eac8e5a085062     MHQ Code Revision: 28c61bbccb0f3c650181c33b74ad2bb654912eec

### Attach Files

[Undeploy Bug.zip](https://github.com/user-attachments/files/18283830/Undeploy.Bug.zip)

### Final Checklist

- [X] I've checked to make sure that this issue has not already been filed
- [X] I'm reporting only one issue in this ticket for clarity and focus

**Comment by Scoppio** [comment]
Yeah, it is very messed up the GM deploy/undeploy, it should only be used if you are willing to deal with the consequences...

**Comment by IllianiCBT** [comment]
GM Undeploy is included begrudgingly due to user request. However is provided with no insurance that it won't cause issues. Flagging as _Won't Fix_ but no closing, in case someone decides to address the numerous issues at some point in the future.

**Comment by IllianiCBT** [comment]
Closing as no longer relevant
